scripts/deleteSnapsets.js

```diff
@@ -0,0 +1,243 @@
+dbLoader.gpiiKeys = [];
+
+fluid.log("COUCHDB_URL: '" +
+    dbLoader.parsedCouchDbUrl.protocol + "//" +
+    dbLoader.parsedCouchDbUrl.hostname + ":" +
+    dbLoader.parsedCouchDbUrl.port +
+    dbLoader.parsedCouchDbUrl.pathname +
+"'");
+
+/**
+ * Find the Prefs Safes of type "snapset", mark them to be deleted and add
+ * them to an array of records to remove.
+ * @param {String} responseString - The response from the database query for
+ *                                  retrieving the snapset PrefsSafes records
+ * @return {Array} - Array of snapset Prefs Safes each with their "deleted"
+ *                   field set to true.
+ */
+dbLoader.processSnapsets = function (responseString) {
+    fluid.log("Processing the snapset Prefs Safes records...");
+    dbLoader.snapSets = JSON.parse(responseString);
+    fluid.each(dbLoader.snapSets.rows, function (aSnapset) {
+        aSnapset.value._deleted = true;
+        dbLoader.snapsetPrefsSafes.push(aSnapset.value);
+    });
+    fluid.log("\tSnapset Prefs Safes marked for deletion.");
+    return dbLoader.snapsetPrefsSafes;
+};
+
+/**
+ * Find the GPII Key records that are associated with a snapset PrefsSafe, mark
+ * them for deletion, and add them to array of records to delete.
+ * @param {String} responseString - The response from the database query for
+ *                                  retrieving all the GPII Keys.
+ * @return {Array} - Array of snapset PrefsSafes' GPII Keys with the "deleted"
+ *                   field set to true.
+ */
+dbLoader.processGpiiKeys = function (responseString) {
+    fluid.log("Processing the GPII Keys...");
+    var gpiiKeyRecords = JSON.parse(responseString);
+    dbLoader.gpiiKeys = dbLoader.findAndDeletePrefsSafesGpiiKeys(
+        gpiiKeyRecords, dbLoader.snapsetPrefsSafes
+    );
+    fluid.log("\tGPII Keys associated with snapset Prefs Safes marked for deletion.");
+    return dbLoader.gpiiKeys;
+};
+
+/**
+ * Given all the GPII Keys records in the database, find the ones that reference
+ * a snapset PrefsSafe.  As each GPII Key is found it is marked for
+ * deletion.
+ * @param {Array} gpiiKeyRecords - Array of GPII Key records from the database.
+ * @return {Array} - the values from the gpiiKeyRecords that are snapset GPII Keys.
+ */
+dbLoader.findAndDeletePrefsSafesGpiiKeys = function (gpiiKeyRecords, snapSets) {
+    var gpiiKeysToDelete = [];
+    fluid.each(gpiiKeyRecords.rows, function (gpiiKeyRecord) {
+        var gpiiKey = fluid.find(snapSets, function (aSnapSet) {
+            if (gpiiKeyRecord.value.prefsSafeId === aSnapSet._id) {
+                return gpiiKeyRecord.value;
+            }
+        }, null);
+        if (gpiiKey !== null) {
+            gpiiKey._deleted = true;
+            gpiiKeysToDelete.push(gpiiKey);
+        }
+    });
+    return gpiiKeysToDelete;
+};
+
+/**
+ * Log that the batch deletion of snapset Prefs Safes and their GPII Keys has
+ * been completed.
+ */
+dbLoader.procesBatchDelete = function () {
```

Comment:
Typo: proces -> process

Reply:
I've already removed the function and replaced it with an anonymous one since it's just a log message.